src/MagnumPlugins/DrFlacAudioImporter/DrFlacImporter.cpp

```diff
@@ -51,21 +87,14 @@ void DrFlacImporter::doOpenData(Containers::ArrayView<const char> data) {
     }
 
     uint64_t samples = handle->totalSampleCount;
-    uint32_t frequency = handle->sampleRate;
     uint8_t numChannels = handle->channels;
     uint8_t bitsPerSample = handle->bitsPerSample;
 
-    _frequency = frequency;
-
-    if(numChannels == 1 && bitsPerSample == 8)
-        _format = Buffer::Format::Mono8;
-    else if(numChannels == 1 && bitsPerSample == 16)
-        _format = Buffer::Format::Mono16;
-    else if(numChannels == 2 && bitsPerSample == 8)
-        _format = Buffer::Format::Stereo8;
-    else if(numChannels == 2 && bitsPerSample == 16)
-        _format = Buffer::Format::Stereo16;
-    else {
+    // Normalize bit amounts to multiples of 8, rounding up
+    UnsignedInt normalizedBitsPerSample = (bitsPerSample / 8) + ((bitsPerSample % 8) ? 1 : 0);
```

Comment:
maybe `normalizedBytesPerSample` would match the actual value better? Also, shortening the code:

```cpp
UnsignedInt normalizedBytesPerSample = (bitsPerSample + 7)/8;
```

Also, what are the allowed values of `bitsPerSample` in the FLAC format? Can you put that into a comment?